### PR TITLE
feat: queue scope and mark event publication

### PR DESCRIPTION
**Pull request by willkill07**
#### Overview

Keep scope and mark emission synchronous while moving event sanitization and subscriber publication onto a serial background dispatcher.

- [x] I confirm this contribution is my own work, or I have the right to submit it under this project's license.
- [x] I searched existing issues and open pull requests, and this does not duplicate existing work.

#### Details

- Snapshot events, visible sanitizer chains, and subscribers at emission time.
- Process snapshots through a FIFO background queue so start, end, and mark ordering is preserved.
- Fail open on sanitizer errors and make dispatcher send failures observable.
- Document delayed publication semantics.

This is PR 1 of 2 in the RELAY-509 GitHub stack, which is rooted on upstream `main`:

1. **#570 — queued event publication**
2. [#571 — async middleware across primary bindings](https://github.com/NVIDIA/NeMo-Relay/pull/571), based on #570

#### Where should the reviewer start?

Start with `crates/core/src/api/runtime/subscriber_dispatcher.rs` and the FIFO/snapshot coverage in `crates/core/tests/integration/subscriber_dispatcher_tests.rs`.

Validation:

- `cargo test -p nemo-relay --test subscriber_dispatcher_integration -- --nocapture`
- Commit hooks, including Cargo fmt, clippy, check, and docs link validation

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- Relates to RELAY-509

## Summary by CodeRabbit

* **New Features**
  * Subscriber delivery can now apply an optional sanitizer chain before invoking callbacks, snapshotted at emission time to preserve scope/mark ordering.
  * Native dispatch supports delivering events with an explicit sanitizer list and scope context.
  * Node.js `flushSubscribers()` is now Promise-based (should be `await`ed).
* **Bug Fixes**
  * Sanitizer panics are caught and logged; delivery continues with the last valid sanitized snapshot.
  * If there are no subscribers, sanitizer callbacks are not invoked.
* **Documentation**
  * Added “Publication Semantics” describing sanitizer timing and FIFO behavior.
* **Tests**
  * Updated integration and Node.js tests to await `flushSubscribers()` and validate sanitizer edge cases.

### CODE REVIEW

**Comment by willkill07** [comment]
@coderabbitai review

**Review by bbednarski9 — APPROVED** [review]

**Comment by willkill07** [comment]
@coderabbitai review

**Comment by willkill07** [comment]
Addressed both outside-diff CodeRabbit findings in 7e5e0bc6: unsanitized delivery now avoids cloning the event, and an integration test verifies sanitizer panics publish the original snapshot.

**Comment by willkill07** [comment]
@coderabbitai review

File: crates/core/src/api/runtime/subscriber_dispatcher.rs [hunk 1]
```diff
@@ -160,12 +204,29 @@ mod native {
...
+            }))
+            .unwrap_or_else(|_| {
+                log::error!(
+                    target: "nemo_relay.runtime",
+                    event = "event_sanitizer_panicked";
+                    "Event sanitizer panicked; publishing the original event snapshot"
+                );
+                original
+            })
+        };
```
[the remark this answers is not in the document]
Comment (willkill07) [reply crates/core/src/api/runtime/subscriber_dispatcher.rs]: Fixed in 5e8e14fd. Each sanitizer is now panic-isolated with its guardrail name logged, and the regression test verifies an earlier redaction survives a later sanitizer panic.

**Comment by willkill07** [comment]
@coderabbitai review

**Review by mnajafian-nv — APPROVED** [review]
LGTM

**Comment by willkill07** [comment]
@coderabbitai review

**Review by mnajafian-nv — APPROVED** [review]

**Comment by willkill07** [comment]
@coderabbitai review

**Comment by willkill07** [comment]
@coderabbitai review

File: crates/node/src/api/mod.rs [hunk 2]
```diff
@@ -3262,17 +3262,21 @@ pub fn deregister_subscriber(name: String) -> Result<bool> {
...
+/// JavaScript subscribers are queued through Node's `ThreadsafeFunction`. Awaiting this
+/// Promise does not block the Node event loop while event sanitizers settle.
 #[napi]
-pub fn flush_subscribers() -> Result<()> {
-    core_subscriber_api::flush_subscribers().map_err(to_napi_err)
+pub async fn flush_subscribers() -> Result<()> {
+    tokio::task::spawn_blocking(core_subscriber_api::flush_subscribers)
+        .await
+        .map_err(|error| to_napi_err(FlowError::Internal(error.to_string())))?
+        .map_err(to_napi_err)
```
[the remark this answers is not in the document]
Comment (willkill07) [reply crates/node/src/api/mod.rs:3257]: Fixed in 79a39795. The generated public docs now state both rejection paths and advise callers to handle await errors. While reproducing CI, I also found the OpenClaw live smoke still treated flushSubscribers() as synchronous; its session-close path now awaits the Promise, and the live ATIF export test passes.

**Comment by willkill07** [comment]
@coderabbitai review

**Comment by willkill07** [comment]
@coderabbitai review

**Review by bbednarski9 — APPROVED** [review]
